rcl_action/test/rcl_action/test_action_server.cpp

```diff
@@ -0,0 +1,738 @@
+  EXPECT_EQ(ret, RCL_RET_INVALID_ARGUMENT) << rcl_get_error_string().str;
+  rcl_reset_error();
+
+  // Initialize with a null name
+  ret = rcl_action_server_init(&action_server, &node, ts, nullptr, &options);
+  EXPECT_EQ(ret, RCL_RET_INVALID_ARGUMENT) << rcl_get_error_string().str;
+  rcl_reset_error();
+
+  // Initialize with a null options
+  ret = rcl_action_server_init(&action_server, &node, ts, action_name, nullptr);
+  EXPECT_EQ(ret, RCL_RET_INVALID_ARGUMENT) << rcl_get_error_string().str;
+  rcl_reset_error();
+
+  // Initialize with valid arguments
+  ret = rcl_action_server_init(&action_server, &node, ts, action_name, &options);
+  EXPECT_EQ(ret, RCL_RET_OK) << rcl_get_error_string().str;
+
+  // Try to initialize again
+  ret = rcl_action_server_init(&action_server, &node, ts, action_name, &options);
+  EXPECT_EQ(ret, RCL_RET_ALREADY_INIT) << rcl_get_error_string().str;
+  rcl_reset_error();
+
+  // Finalize with null action server
+  ret = rcl_action_server_fini(nullptr, &node);
+  EXPECT_EQ(ret, RCL_RET_ACTION_SERVER_INVALID) << rcl_get_error_string().str;
+  rcl_reset_error();
+
+  // Finalize with invalid action server
+  rcl_action_server_t invalid_action_server = rcl_action_get_zero_initialized_server();
+  ret = rcl_action_server_fini(&invalid_action_server, &node);
+  // Nothing happens
+  EXPECT_EQ(ret, RCL_RET_OK) << rcl_get_error_string().str;
+
+  // Finalize with null node
+  ret = rcl_action_server_fini(&action_server, nullptr);
+  EXPECT_EQ(ret, RCL_RET_NODE_INVALID) << rcl_get_error_string().str;
+  rcl_reset_error();
+
+  // Finalize with valid arguments
+  ret = rcl_action_server_fini(&action_server, &node);
+  EXPECT_EQ(ret, RCL_RET_OK) << rcl_get_error_string().str;
+
+  // Finalize node
+  ret = rcl_node_fini(&node);
+  EXPECT_EQ(RCL_RET_OK, ret) << rcl_get_error_string().str;
+
+  ret = rcl_shutdown();
+  EXPECT_EQ(RCL_RET_OK, ret) << rcl_get_error_string().str;
+}
+
+class TestActionServer : public ::testing::Test
+{
+protected:
+  void SetUp()
+  {
+    rcl_ret_t ret = rcl_init(0, nullptr, rcl_get_default_allocator());
+    ASSERT_EQ(RCL_RET_OK, ret) << rcl_get_error_string().str;
+    this->node = rcl_get_zero_initialized_node();
+    rcl_node_options_t node_options = rcl_node_get_default_options();
+    ret = rcl_node_init(&this->node, "test_action_server_node", "", &node_options);
+    ASSERT_EQ(RCL_RET_OK, ret) << rcl_get_error_string().str;
+    const rosidl_action_type_support_t * ts = ROSIDL_GET_ACTION_TYPE_SUPPORT(
+      test_msgs, Fibonacci);
+    const rcl_action_server_options_t options = rcl_action_server_get_default_options();
+    const char * action_name = "test_action_server_name";
+    this->action_server = rcl_action_get_zero_initialized_server();
+    ret = rcl_action_server_init(&this->action_server, &this->node, ts, action_name, &options);
+    ASSERT_EQ(ret, RCL_RET_OK) << rcl_get_error_string().str;
+  }
+
+  void TearDown()
```

Comment:
@jacobperron missing trailing `override`.

Reply:
Added in 69c0287